common.hpp

```diff
@@ -17,276 +17,301 @@
-                                                                        //UNLIB_DEFINE_SCALED_LITERAL(Qty_,atto , a##UnitShort_)
-/** define the literal operators for all the macro unit prefixes (deca-tera) */
-#define UNLIB_DEFINE_METRIC_MACRO_PREFIXED_LITERAL(Qty_,UnitShort_)       UNLIB_DEFINE_SCALED_LITERAL(Qty_,deca ,dk##UnitShort_) \
-                                                                          UNLIB_DEFINE_SCALED_LITERAL(Qty_,hecto, h##UnitShort_) \
-                                                                          UNLIB_DEFINE_SCALED_LITERAL(Qty_,kilo , k##UnitShort_) \
-                                                                          UNLIB_DEFINE_SCALED_LITERAL(Qty_,mega , M##UnitShort_) \
-                                                                          UNLIB_DEFINE_SCALED_LITERAL(Qty_,giga , G##UnitShort_) \
-                                                                          UNLIB_DEFINE_SCALED_LITERAL(Qty_,tera , T##UnitShort_)
-                                                                        //UNLIB_DEFINE_SCALED_LITERAL(Qty_,peta , P##UnitShort_)
-                                                                        //UNLIB_DEFINE_SCALED_LITERAL(Qty_,exa  , E##UnitShort_)
-/** define the literal operators for all unit prefixes (atto-tera) */
-#define UNLIB_DEFINE_METRIC_PREFIXED_LITERALS(Qty_,UnitShort_)            UNLIB_DEFINE_METRIC_MICRO_PREFIXED_LITERAL(Qty_,UnitShort_) \
-                                                                          UNLIB_DEFINE_METRIC_MACRO_PREFIXED_LITERAL(Qty_,UnitShort_)
-/** define the literal operators for a physical unit (nor prefixes) */
-#define UNLIB_DEFINE_LITERAL(Qty_,UnitShort_)                             UNLIB_DEFINE_SCALED_LITERAL(Qty_,no_scale,UnitShort_)
-/** @} */
-
-
 /**
  * @{
  *
- * time quantities
+ * quantities of basic units
  *
  * @tparam V  value type
  */
-template<typename V> using   second = quantity<time, no_scaling, V>;                                             UNLIB_DEFINE_LITERAL(second, s   )            UNLIB_DEFINE_METRIC_PREFIXED_LITERALS(second, s)
-template<typename V> using   minute = minute_scale<second<V>>;                                                   UNLIB_DEFINE_LITERAL(minute, min )
-template<typename V> using     hour =   hour_scale<second<V>>;                                                   UNLIB_DEFINE_LITERAL(  hour, h   )
-template<typename V> using      day =    day_scale<second<V>>;                                                   UNLIB_DEFINE_LITERAL(   day, d   )
-template<typename V> using     week =   week_scale<second<V>>;                                                   UNLIB_DEFINE_LITERAL(  week, week)
+template<typename V> using        second = quantity<time              , no_scaling   , V>;
+template<typename V> using          gram = quantity<mass              , milli_scaling, V>; /**< @note Base unit of mass is not gram, but kilogram. */
```

Comment:
see my other comment regarding the handling of (kilo)gram.

in any case the base unit defined here should IMO be `kilogram`

Reply:
👍🏻